redis/src/test/java/org/apache/calcite/adapter/redis/RedisITCaseBase.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.adapter.redis;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+
+import redis.embedded.RedisServer;
+
+/**
+ * RedisITCaseBase.
+ */
+public abstract class RedisITCaseBase {
+
+  public static final int PORT = getAvailablePort();
+  public static final String HOST = "127.0.0.1";
+  private static final String MAX_HEAP = "maxheap 51200000";
+
+  private static RedisServer redisServer;
+
+  @BeforeClass
+  public static void createRedisServer() throws IOException {
+    if (isWindows()) {
```

Comment:
Will it be easier to use pure java implementation of redis server like [redis-mock](https://github.com/fppt/jedis-mock) ? 

This way there is no need to keep track of temporary files, open ports etc.

Reply:
> Will it be easier to use pure java implementation of redis server like [redis-mock](https://github.com/fppt/jedis-mock) ?
> 
> This way there is no need to keep track of temporary files, open ports etc.

I have tested redis-mock and some types of operations are not supported.